pkg/apis/istio/v1alpha2/validation/validation_test.go

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package validation
+
+import (
+	"reflect"
+	"testing"
+
+	"istio.io/operator/pkg/apis/istio/v1alpha2"
+)
+
+func makeBoolPtr(v bool) *bool {
```

Comment:
nit: it's more idiomatic in Go no to pre-declare functions, but rather to put them after the function where they are called so the reading flow is top down.

Reply:
moved